libs/pika/execution_base/tests/unit/any_sender.cpp

```diff
@@ -696,6 +626,93 @@ void test_when_all()
     tt::sync_wait(std::move(as3));
 }
 
+// To control when the move constructor will throw, exception is caught in the set_value of
+// any_receiver
+bool throwing = false;
+
+struct throwing_constructor
+{
+    throwing_constructor() noexcept(false) {}
+    throwing_constructor(throwing_constructor&&) noexcept(false)
+    {
+        if (throwing) throw std::runtime_error("error");
+    };
+    // to allow use with any_sender
+    throwing_constructor(throwing_constructor const&) = default;
+    throwing_constructor(throwing_constructor&) noexcept(false)
+    {
+        if (throwing) throw std::runtime_error("error");
+    };
+};
+
+void test_throwing_constructor()
+{
+    {
+        // Throwing move with unique_any_sender
+        throwing = false;
+        std::atomic<bool> set_error_called{false};
+        ex::unique_any_sender<throwing_constructor> as1 =
+            ex::just(std::move(throwing_constructor()));
+        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
+            check_exception_ptr, set_error_called};
+        auto os = ex::connect(std::move(as1), std::move(r));
+        throwing = true;
+        ex::start(os);
+        PIKA_TEST(set_error_called);
+    }
+
+    {
+        // Throwing move with any_sender
+        throwing = false;
+        std::atomic<bool> set_error_called{false};
+        ex::any_sender<throwing_constructor> as1 = ex::just(std::move(throwing_constructor()));
+        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
```

Comment:
Valid warnings triggered by clang:
```suggestion
            ex::just(throwing_constructor());
        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
            check_exception_ptr, set_error_called};
        auto os = ex::connect(std::move(as1), std::move(r));
        throwing = true;
        ex::start(os);
        PIKA_TEST(set_error_called);
    }

    {
        // Throwing move with any_sender
        throwing = false;
        std::atomic<bool> set_error_called{false};
        ex::any_sender<throwing_constructor> as1 = ex::just(throwing_constructor());
        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
```